src/core/Akka.Remote/Remoting.cs

```diff
@@ -186,35 +186,29 @@ public override void Start()
 
                 try
                 {
-                    var addressPromise = new TaskCompletionSource<IList<ProtocolTransportAddressPair>>();
+                    var addressPromise = new TaskCompletionSource<IList<ProtocolTransportAddressPair>>(TaskCreationOptions.RunContinuationsAsynchronously);
 
                     // tells the EndpointManager to start all transports and bind them to listenable addresses, and then set the results
                     // of this promise to include them.
-                    _endpointManager.Tell(new EndpointManager.Listen(addressPromise)); 
+                    _endpointManager.Tell(new EndpointManager.Listen(addressPromise));
 
                     addressPromise.Task.Wait(Provider.RemoteSettings.StartupTimeout);
                     var akkaProtocolTransports = addressPromise.Task.Result;
-                    if(akkaProtocolTransports.Count==0)
+                    if (akkaProtocolTransports.Count == 0)
                         throw new ConfigurationException(@"No transports enabled under ""akka.remote.enabled-transports""");
-                    _addresses = new HashSet<Address>(akkaProtocolTransports.Select(a => a.Address));
```

Comment:
Pardon my ignorance, but where does this get set if not here?

Reply:
3 lines below